universal_transfer_operator/src/universal_transfer_operator/data_providers/database/base.py

```diff
@@ -186,24 +185,19 @@ def check_if_transfer_supported(self, source_dataset: Table) -> bool:
         source_connection_type = get_dataset_connection_type(source_dataset)
         return Location(source_connection_type) in self.transfer_mapping
 
-    def read(self):
+    def read(self) -> Iterator[pd.DataFrame]:
         """Read the dataset and write to local reference location"""
-        with NamedTemporaryFile(mode="wb+", suffix=".parquet", delete=False) as tmp_file:
-            df = self.export_table_to_pandas_dataframe()
-            df.to_parquet(tmp_file.name)
-            local_temp_file = FileStream(
-                remote_obj_buffer=tmp_file.file,
-                actual_filename=tmp_file.name,
-                actual_file=File(path=tmp_file.name),
-            )
-            yield local_temp_file
+        yield self.export_table_to_pandas_dataframe()
 
-    def write(self, source_ref: FileStream):
+    def write(self, source_ref: FileStream | pd.DataFrame):
         """
         Write the data from local reference location to the dataset.
         :param source_ref: Stream of data to be loaded into output table.
         """
-        return self.load_file_to_table(input_file=source_ref.actual_file, output_table=self.dataset)
+        if isinstance(source_ref, FileStream):
```

Comment:
We should comment on the logic of why we are doing the check for both `Filestream` and pandas data frame. Filestream is needed as well for data in formats like zip, images etc and pandas dataframe is relevant for filestypes like csv, json, parquet etc.

Reply:
Yup, added it.